src/Collections-Abstract/Collection.class.st

```diff
@@ -1183,6 +1183,16 @@ Collection >> select: aBlock [
 	^newCollection
 ]
 
+{ #category : #enumerating }
+Collection >> select: aBlock as: aClass [
+	"Evaluate aBlock with each of the receiver's elements as argument.
+	Collect the elements for which aBlock evaluates to true in an instance of aClass. Answer the resulting collection"
+	
+	| selectedElements |
+	selectedElements := (self select: aBlock).
```

Comment:
Why not use `Object>>as:` ?
So this can be 
`^(self select: aBlock) as: aClass` and will avoid to fill the collection if it is of the same type.

Reply:
I tried this replacement, but the image freezes when I save the new #select:as: method.
Maybe because it's used by the announcer system, which tries to announce that this method was changed, but this method fails?